### DISCUSSION TITLE
bug: Firebase dynamiclink event not fired in ios

### DISCUSSION THREAD

**Discussion by karen1au** [ionic-team/capacitor#3658] [Q&A]
## Bug Report
I'm using the ```cordova-plugin-firebase-dynamiclinks``` plugin and it works smoothly in Android, however in iOS i got this in the log:
```
 [Firebase/Analytics][I-ACS023001] Deep Link does not contain valid required params. URL params: {
    amv = "";
    apn = "com.example.app";
    ibi = "com.example.app";
    ifl = "https://exampleproject.firebaseapp.com/__/auth/action?apiKey=xxxxxxxx&mode=signIn&oobCode=xxxxxxxxxxx&continueUrl=https://example.page.link&lang=en";
    link = " "https://exampleproject.firebaseapp.com/__/auth/action?apiKey=xxxxxxxx&mode=signIn&oobCode=xxxxxxxxxxx&continueUrl=https://example.page.link&lang=en";
}
```
FirebaseDynamicLinksPlugin is init, and onDynamicLink was called on launch.
In debugger, it seems that ```swizzled_application:(UIApplication *)app openURL``` and ```swizzled_application:(UIApplication *)app continueUserActivity``` were never triggered, I'm trying to follow the flow of handling the dynamic link  but since I'm not an iOS developer it's quite confusing for me.

I'm not sure if this is the right place to report/ask or if I should post this to ionic-native or the original cordova plugin repo... please let me know if it's not related to capacitor, thanks!

### Capacitor Version

```
  @capacitor/cli: 2.4.2
  @capacitor/core: 2.4.2
  @capacitor/android: 2.4.2
  @capacitor/electron: 2.4.2
  @capacitor/ios: 2.4.2
```

### Platform(s)
iOS

### Current Behavior
When launch app in XCode, FirebaseDynamicLink plugin is init and onDynamicLink is called, clicking the url from browser, it doesn't open the link in the app. After I configured associated domains with ```applinks:example.page.link``` it does open the app (since this part is not in the doc for the plugin, i'm not sure if it's actually required?) but the callback is not executed, in the log I see  ``` [Firebase/Analytics][I-ACS023001] Deep Link does not contain valid required params. ```

### Expected Behavior
Clicking on link in browser should redirect back to app and continue with the sign in process (i'm using dynamic link for authentication)

### Other Technical Details

`npm --version` output: 6.7.0

`node --version` output: v11.14.0

`pod --version` output (iOS issues only): 1.9.3

**Comment by LiLeiAndroid** [comment]
I also have this issue, I can use the deep link open my app in safari browser, but I can not get the url  from " App.addListener('appUrlOpen', (urlOpen: any) ",  and it works perfect on Android , maybe it relate to IOS14.

**Comment by jcesarmobile** [comment]
The plugin has a category on AppDelegate.m file, but Capacitor has AppDelegate.swift, so the code on the category is not executed.
You'll have to check what the plugin does and try to execute the same code on your AppDelegate.swift

https://github.com/chemerisuk/cordova-plugin-firebase-dynamiclinks/blob/master/src/ios/AppDelegate%2BFirebaseDynamicLinksPlugin.m
